src/Propel/Runtime/Util/PropelDateTime.php

```diff
@@ -70,34 +71,53 @@ protected static function isTimestamp($value): bool
      * Usually `new \Datetime()` does not contain milliseconds so you need a method like this.
      *
      * @param string|null $time Optional, in seconds. Floating point allowed.
+     * @param string $dateTimeClass Optional, class name of the created object.
      *
      * @throws \InvalidArgumentException
      *
-     * @return \DateTime
+     * @return \DateTimeInterface An instance of $dateTimeClass
      */
-    public static function createHighPrecision(?string $time = null): DateTime
+    public static function createHighPrecision(?string $time = null, string $dateTimeClass = 'DateTime'): DateTimeInterface
     {
-        $dateTime = DateTime::createFromFormat('U.u', $time ?: self::getMicrotime());
+        $allowStringIsA = true;
+
+        if (!is_a($dateTimeClass, DateTime::class, $allowStringIsA) && !is_a($dateTimeClass, DateTimeImmutable::class, $allowStringIsA)) {
```

Comment:
Mb we can check against interface and not 2 specific classes?

Reply:
The interface doesn't have the `createFromFormat` method:
https://www.php.net/manual/en/class.datetimeinterface.php

I guess using is_a() is the only possibility?